test/test_images/probe_helper/probe/cloudscheduler.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"context"
+	"fmt"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+)
+
+const (
+	// CloudSchedulerSourceProbeEventType is the CloudEvent type of forward
+	// CloudSchedulerSource probes.
+	CloudSchedulerSourceProbeEventType = "cloudschedulersource-probe"
+)
+
+// CloudSchedulerSourceProbe is the probe handler for probe requests in the
+// CloudSchedulerSource probe.
+type CloudSchedulerSourceProbe struct{}
+
+var cloudSchedulerSourceProbe Handler = &CloudSchedulerSourceProbe{}
+
+func (p *CloudSchedulerSourceProbe) Forward(ctx context.Context, ph *Helper, event cloudevents.Event) error {
+	// Get the receiver channel
+	channelID := fmt.Sprintf("%s/%s", event.Extensions()[probeEventTargetServiceExtension], "cloudschedulersource-probe")
+	receiverChannel, cleanupFunc, err := CreateReceiverChannel(ctx, ph, channelID)
```

Comment:
How do we deal with the static scheduler source now? So it will create and destroy the channels? What happens in case of concurrent probing?

Reply:
This is a work in progress, the current implementation on this PR is not complete. I think it makes sense to keep a map of timestamps in the receiver, check that map during probes in the forward handlers, and leverage the liveness probe to garbage collect the stale timestamps. I will work on this.

Reply:
There should be no problem with concurrent probing now. All different scheduler tick times are tracked, and during liveness probes, the stale ticks (>3 minutes old) are discarded from being tracked.